### PR TITLE
i#3372: Fixes eflag restoration bug

### PR DESCRIPTION
**Pull request by johnfxgalea**
Checks whether a written reg, particularly XAX, is being used to store eflag data. If this is the case, we enter the branch so we can move eflags from the register.

This PR fixes issue #3372.

### CODE REVIEW

**Comment by johnfxgalea** [comment]
Done. LMKWYT

File: suite/tests/client-interface/drreg-test.c [hunk 1]
```diff
@@ -274,7 +274,21 @@ GLOBAL_LABEL(FUNCNAME:)
...
+        /* Store aflags to dead XAX, and restore when XAX is live */
+     test11:
+        mov      TEST_REG_ASM, DRREG_TEST_11_ASM
+        mov      TEST_REG_ASM, DRREG_TEST_11_ASM
+        cmp      TEST_REG_ASM, TEST_REG_ASM
+        push     TEST_CONST
+        pop      REG_XAX
+        mov      REG_XAX, TEST_REG_ASM
+        mov      TEST_REG_ASM, REG_XAX
+        je       epilog
```
[the remark this answers is not in the document]
Comment (johnfxgalea) [reply suite/tests/client-interface/drreg-test.c]: Yep, definitely agree with you there. I'll get that sorted tonight.

[the remark this answers is not in the document]
Comment (johnfxgalea) [reply suite/tests/client-interface/drreg-test.c]: Done.

**Comment by derekbruening** [comment]
It looks like it's failing the clang-format check.  Maybe you have a different version of clang-format? We're using 6.0.  If you could make it match via the changes here: https://travis-ci.com/DynamoRIO/dynamorio/jobs/177389541

**Comment by johnfxgalea** [comment]
Done.

**Pull request merged**